src/layout.cpp

```diff
@@ -436,6 +436,12 @@ int frame_current_bring(int argc, char** argv, Output output) {
     }
     HSTag* tag = get_current_monitor()->tag;
     global_tags->moveClient(client, tag, {}, true);
+    auto frame = tag->frame->root_->frameWithClient(client);
+    if (!client->is_client_floated() && !frame->isFocused()) {
+        frame->removeClient(client);
+        tag->frame->focusedFrame()->insertClient(client);
+    }
+    focus_client(client, false, false, true);
```

Comment:
Instead of `focus_client()` I think it's enough to pass `focus = true` to `insertClient()`.

Reply:
I hope it was OK that I've implemented this directly.

Reply:
This does seem to break it for me again. The brought client does not get focus and is still dislayed in a different frame. If I cycle through them, then it's frame gets updated. I'll remove your change again and update the test accordingly.